src/main/java/org/cryptomator/frontend/fuse/mount/AbstractMounter.java

```diff
@@ -0,0 +1,62 @@
+package org.cryptomator.frontend.fuse.mount;
+
+import org.cryptomator.frontend.fuse.AdapterFactory;
+import org.cryptomator.frontend.fuse.FuseNioAdapter;
+
+import java.nio.file.Path;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+public abstract class AbstractMounter implements Mounter {
+
+    private static final int MOUNT_TIMEOUT_MILLIS = 3000;
+    private static final AtomicInteger MOUNT_COUNTER = new AtomicInteger(0);
+
+    @Override
+    public synchronized Mount mount(Path directory, EnvironmentVariables envVars, Consumer<Throwable> onFuseExit, boolean debug) throws FuseMountException {
+        CountDownLatch mountSuccessSignal = new CountDownLatch(1);
+        AtomicReference<Throwable> exception = new AtomicReference<>();
+        FuseNioAdapter fuseAdapter = AdapterFactory.createReadWriteAdapter(directory, //
+                AdapterFactory.DEFAULT_MAX_FILENAMELENGTH, //
+                envVars.getFileNameTranscoder());
+        MountSuccessFuseDecorator decoratedFuseAdapter = new MountSuccessFuseDecorator(fuseAdapter, mountSuccessSignal);
+        //real mount op
+        var mountThread = new Thread(() -> {
+            try {
+                decoratedFuseAdapter.mount(envVars.getMountPoint(), true, debug, envVars.getFuseFlags());
+            } catch (Exception e) {
+                exception.set(e);
+            } finally {
+                onFuseExit.accept(exception.get());
+            }
+        });
+        mountThread.setName("fuseMount-" + MOUNT_COUNTER.getAndIncrement() + "-main");
+        mountThread.setDaemon(true);
+        mountThread.start();
+
+        // wait for mounted() is called, unlocking the barrier
+        try {
+            if (!mountSuccessSignal.await(MOUNT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS)) {
+                if (exception.get() != null) {
+                    throw new FuseMountException(exception.get());
+                }
+                throw new FuseMountException("Mounting operation timed out");
```

Comment:
wrap in else, so one can immediately see that both branches are mutually exclusive

Reply:
Fixed in 72684a2